### ISSUE TITLE
Issue creating lambda layer version due to key not present

### ISSUE THREAD

**Issue by gitamolsrivastav** [aws-ia/terraform-aws-control_tower_account_factory#27]
I am facing issue creating a resource due to non existent s3 key. Could you please help identifying the cause here? My guess its not referencing the path correctly
```
module.aft.module.aft_lambda_layer.aws_lambda_layer_version.layer_version: Creating...
╷
│ Error: Error creating lambda layer: InvalidParameterValueException: Error occurred while GetObject. S3 Error Code: NoSuchKey. S3 Error Message: The specified key does not exist.
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "7980129f-6f43-4ae7-9952-99df5af92aec"
│   },
│   Message_: "Error occurred while GetObject. S3 Error Code: NoSuchKey. S3 Error Message: The specified key does not exist.",
│   Type: "User"
│ }
│ 
│   with module.aft.module.aft_lambda_layer.aws_lambda_layer_version.layer_version,
│   on .terraform/modules/aft/modules/aft-lambda-layer/main.tf line 24, in resource "aws_lambda_layer_version" "layer_version":
│   24: resource "aws_lambda_layer_version" "layer_version" {
│ 
╵
```
```
resource "aws_lambda_layer_version" "layer_version" {
  lifecycle {
    create_before_destroy = true
  }

  layer_name          = "${var.lambda_layer_name}-${replace(time_sleep.lambda_layer_wait.triggers["lambda_layer_version"], ".", "-")}"
  compatible_runtimes = ["python${var.lambda_layer_python_version}"]
  s3_bucket           = var.s3_bucket_name
  s3_key              = "layer.zip"
}
```

document referred while creating resources: https://learn.hashicorp.com/tutorials/terraform/aws-control-tower-aft

**Comment by shmykyta** [comment]
@gitamolsrivastav Hey, had same issue for me, use the latest version, that's fix in 1.0.11
`source = "github.com/aws-ia/terraform-aws-control_tower_account_factory?ref=1.0.11"`

**Comment by gitamolsrivastav** [comment]
^^ version update did not fix the issue. I did
```module "aft" {
  source = "github.com/aws-ia/terraform-aws-control_tower_account_factory?ref=1.0.11"
  ct_management_account_id    = var.ct_management_account_id
  log_archive_account_id      = var.log_archive_account_id
  audit_account_id            = var.audit_account_id
  aft_management_account_id   = var.aft_management_account_id
  ct_home_region              = var.ct_home_region
  tf_backend_secondary_region = var.tf_backend_secondary_region

  vcs_provider                                  = "github"
  account_request_repo_name                     = "${var.github_username}/learn-terraform-aft-account-request"
  account_provisioning_customizations_repo_name = "${var.github_username}/learn-terraform-aft-account-provisioning-customizations"
  global_customizations_repo_name               = "${var.github_username}/learn-terraform-aft-global-customizations"
  account_customizations_repo_name              = "${var.github_username}/learn-terraform-aft-account-customizations"
}
```
terraform init --> terraform plan -->terraform apply

but got the same non-existent key error

```
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.aft.module.aft_lambda_layer.aws_lambda_layer_version.layer_version: Creating...
╷
│ Error: Error creating lambda layer: InvalidParameterValueException: Error occurred while GetObject. S3 Error Code: NoSuchKey. S3 Error Message: The specified key does not exist.
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "58c66a1f-70ba-4383-baeb-1f41a5254963"
│   },
│   Message_: "Error occurred while GetObject. S3 Error Code: NoSuchKey. S3 Error Message: The specified key does not exist.",
│   Type: "User"
│ }
│ 
│   with module.aft.module.aft_lambda_layer.aws_lambda_layer_version.layer_version,
│   on .terraform/modules/aft/modules/aft-lambda-layer/main.tf line 24, in resource "aws_lambda_layer_version" "layer_version":
│   24: resource "aws_lambda_layer_version" "layer_version" {
│ 
```

**Comment by gitamolsrivastav** [comment]
![image](https://user-images.githubusercontent.com/22173427/149252425-4d5beaef-7346-4742-ad40-6706b203b0f9.png)

**Comment by gitamolsrivastav** [comment]
so I was able to resolve the issue by making sure my account cscore is good. I created few instances in my management account and ran the terraform apply again.

I was getting above key error due to aws codebuild not able to build the project and the error in codebuild was
```
Build failed to start
Build failed to start. The following error occurred: Cannot have more than 0 builds in queue for the account

basically account not having privilege to run code builds
```
which lead me to cscore discussion with aws support.

**Comment by witchbutter** [comment]
What is cscore?  Can you provide any more detail?
